#161966897 CRUD tests for article

kamyaD opened the pull request:
#### What does this PR do?

This PR contains failing CRUD tests for articles

#### Description of Task to be completed?

- add a test for creating an article
- add a test for retrieving  an article
- add test for editing an article
- add a test for  deleting an article

#### How should this be manually tested?

- Clone the repo and cd into the project folder
- Create a virtual environment by running `$ python3 -m venv <name_of_environment>`
- Activate the virtual environment by running `source <name_of_environment>/bin/activate`
- Run `$ pip install -r requirements.txt`
- run python manage.py test

#### What are the relevant pivotal tracker stories?

[ #161966897 ](https://www.pivotaltracker.com/story/show/161966897)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/33648833/49390655-92ed7700-f73a-11e8-89a3-926c504f8c29.png)

ThaDeveloper committed:
Merge pull request #6 from andela/ch-login-and-registration-16196689

#161966899 mockup login and registration  pages

kamyaD committed:
chore(article): test for article

- test  whether user can create, read, update and delete article

[Delivers #161966897]

kamyaD committed:
chore(article): test for article

- test  whether user can create, read, update and delete article

[Delivers #161966897]

kamyaD committed:
chore(article): test for article

- test  whether user can create, read, update and delete article

[Delivers #161966897]

kamyaD committed:
chore(article): test for article

- test  whether user can create, read, update and delete article

[Delivers #161966897]

PromasterGuru committed:
chore(configure-postgresql-db): Configure Authors Haven to use PostgreSQL as DB Engine (#16)

- ensure Authors Haven is configured to use PostgreSQL as DB Engine instead of the default sqlite DB

[Delivers #161966905]

Commit:
chore(test-registration): Test user registration

-Add .DS_Store,.vscode and virtual folder to .gitignore
-Create test for user registration
-Fix routing errors
-Add input validation tests
-Add test to check for duplicate data
-Test for missing data, valid username and password
-Add tests for missing parameters
-Configure code to pep8 standards
-Generate user token

[Starts #161966897]

r-wambui commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
Please remove the vscode. Why do we still have this I had talked about it in the last PR

r-wambui commented on authors/settings.py:
```diff
@@ -138,6 +138,6 @@
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
+        # 'authors.apps.authentication.backends.JWTAuthentication',
```
Do not push commented code.

r-wambui commented on authors/urls.py:
```diff
@@ -19,6 +19,6 @@
     url(r'^admin/', admin.site.urls),
 
-    url(r'^api/', include('authors.apps.authentication.urls', namespace='authentication')),
+    url(r'^api/', include('authors.apps.authentication.urls')),
```
We should have this reflect implementation of urls in django 2.

r-wambui commented on authors/apps/tests/test_article.py:
```diff
@@ -0,0 +1,55 @@
+from django.test import TestCase
+# from ..models import Article
```
Since you created the articles model why do we still have this?

kamyaD replied:
Hello Rose, I am still cooking something, my next push will not have these errors. Please bear with me.

codeplus254 commented:
Great work! Remember to squash the commits after you finish cooking :-P so that the review process is easy.

PromasterGuru commented:
@kamyaD resolve the conflicts as you work on the other feedbacks.

kamyaD committed:
Merge branch 'develop' into Ch-unit-tests-161966897

kamyaD commented:
@r-wambui I have just pushed my latest changes which has all the feedbacks that you mentioned. Please have a look and let me know. Thanks

kamyaD commented:
@PromasterGuru  I have resolved the conflicts as you mentioned. Thanks for the ask feedback

kamyaD committed:
fix merge conflict

amutava commented on authors/apps/articles/tests/test_views.py:
```diff
@@ -0,0 +1,76 @@
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    # def test_user_can_update_comments(self):
```
why do you have commented code on your PR?

kamyaD replied:
Worked on this.

PromasterGuru committed:
bug(fix-coveralls): Fix coveralls to display coverage results

- Updated travis.yml file to run coveralls after the build pass

[Finishes #161966907]

PromasterGuru committed:
chore(tag-articles): Implement tests for user can tag articles

- Implement tests for user can create tags
- Implement tests for user can get a specific tag
- Implement tests for user can get all tags
- Implement tests for user can update an tag
- Implement tests for user can delete a tag
- Implement tests for usr can add an article
- Implement tests for user can update an article

[Delivers #161966897]

PromasterGuru committed:
chore(article-rating): Implement tests for article ratings

- Implement failing tests for article rating

[Delivers #161966897]

PromasterGuru committed:
chore(like-articles): Implement tests for article likes or dislikes

- Implement failing tests for article likes and dislikes

[Delivers #161966897]

PromasterGuru committed:
chore(favorite-article): Implement tests for favoriting articles

- Implement failing tests to check whether user can favorite or unfavorite  an article

[Delivers #161966897]

r-wambui commented:
@kamyaD Please rebase develop. I merged your other PR. Also resolve the merge conflicts

amutava commented on authors/apps/articles/tests/test_views.py:
```diff
@@ -0,0 +1,76 @@
+    #         reverse('update_comment'), self.comment_data, format="json")
+    #     comment = Comment.objects.get()
+    #     self.change_article = {'title': 'The love storry'}
+    #     res = self.client.put(
```
remove commented code

leonardnjura commented:
Good work @kamyaD , Though, let's strive for utter dryness so code climate won't sense so much global warming and complain that our Infinite selves are becoming an endangered species

kamyaD committed:
Merge branch 'develop' into Ch-unit-tests-161966897

kamyaD commented:
@r-wambui @andela-amutava I have worked on the feedbacks and rebased tests @PromasterGuru branches to have his tests too so that they could be reflected in this PR.
